modules/ingester/client/client.go

```diff
@@ -45,7 +45,7 @@ func New(addr string, cfg Config) (*Client, error) {
 	opts := []grpc.DialOption{
 		grpc.WithInsecure(),
 		grpc.WithDefaultCallOptions(
-			grpc.UseCompressor("gzip"),
+			grpc.UseCompressor("snappy"),
```

Comment:
Did you intend to use `c.IngesterClient.GRPCClientConfig.GRPCCompression` here?

Reply:
No, as far as I can tell, this default is a backup for a mis-configured ingester_client setting.  In normal cases `c.IngesterClient.GRPCClientConfig.GRPCCompression` is present and overrides this.  However it is possible to enter yaml such as:
```
ingester_client:
   grpc_client_config:
```

that will cause `c.IngesterClient.GRPCClientConfig.GRPCCompression` to be blank, somewhat unexpectedly.

If we remove this default then it means situations like the above would not have any compression (default "")

Reply:
Hrm, ok agree with this change. Had one last question: If someone deliberately leaves the GRPCCompression field blank in the YAML like you mentioned, would we still enable snappy?

Reply:
Upon further testing, the DefaultCallOption should be removed entirely. No compression is a lack of any UseCompressor call options, which is not possible while the default exists.  After removing the DefaultCallOption this does work to use no compression:
```
ingester_client:
   grpc_client_config:
        grpc_compression: "" 
```

Disregard my comment about mis-configured yaml doing this unexpectedly.  It's because in the case of:
```
ingester_client:
   grpc_client_config:
```
`grpc_client_config` is being treated as empty string instead of an object.  It breaks more than the compression and is not useable: min/max message size are 0 and nothing will work.